bluemira/equilibria/coils/_grouping.py

```diff
@@ -0,0 +1,1088 @@
+
+    def n_coils(self, ctype: Optional[Union[str, CoilType]] = None) -> int:
+        """
+        Get number of coils
+
+        Parameters
+        ----------
+        ctype: Optional[Union[str, CoilType]]
+            get number of coils of a specific type
+
+        Returns
+        -------
+        int
+            Number of coils
+        """
+        if ctype is None:
+            return len(self.x)
+
+        if not isinstance(ctype, CoilType):
+            ctype = CoilType[ctype]
+
+        return Counter(self.ctype)[ctype]
+
+    def plot(self, ax=None, subcoil=True, label=False, force=None, **kwarg):
+        """
+        Plot a CoilGroup
+        """
+        return CoilGroupPlotter(
+            self, ax=ax, subcoil=subcoil, label=label, force=force, **kwarg
+        )
+
+    def fix_sizes(self):
+        """
+        Fix coils sizes
+        """
+        for fx in self.__list_getter("fix_size"):
+            fx()
+
+    def add_coil(self, *coils: Union[Coil, CoilGroup[Coil]]):
+        """Add coils to the coil group"""
+        self._coils = (*self._coils, *coils)
+
+    def remove_coil(self, *coil_name: str, _top_level: bool = True) -> Union[None, List]:
+        """
+        Remove coil from CoilGroup
+
+        Parameters
+        ----------
+        coil_name: str
+            coil(s) to remove
+        _top_level: bool
+            FOR INTERNAL USE, flags if at top level of nested coilgroup stack
+
+        Returns
+        -------
+        Union[List, None]
+            Removed names if not at top level of nested stack
+
+        Notes
+        -----
+        If a nested coilgroup is empty it is also removed from the parent coilgroup
+
+        """
+        names = self.name
+
+        if _top_level and (remainder := set(coil_name) - set(names)):
+            raise EquilibriaError(f"Unknown coils {remainder}")
+
+        c_names = []
+        removed_names = []
+
+        for c in self._coils:
+            if isinstance(c, CoilGroup):
+                removed_names.extend(c.remove_coil(*coil_name, _top_level=False))
+            elif c.name in coil_name:
+                c_names.append(c.name)
+
+        to_remove = [names.index(c_n) for c_n in c_names]
+
+        coils = np.array(self._coils, dtype=object)
+        mask = np.full(coils.size, True, dtype=bool)
+        mask[to_remove] = False
+
+        for no, c in enumerate(self._coils):
+            if c.name == []:
+                mask[no] = False
+
+        self._coils = tuple(coils[mask])
+
+        if not _top_level:
+            return removed_names + to_remove
+
+    @classmethod
+    def from_group_vecs(cls, eqdsk: EQDSKInterface):
+        """
+        Initialises an instance of CoilSet from group vectors. This has been
+        implemented as a dict operation, because it will occur for eqdsks only.
+        Future dict instantiation methods will likely differ, hence the
+        confusing name of this method.
```

Comment:
```suggestion
        Initialises an instance of CoilSet from group vectors. 
        
        This has been implemented as a dict operation, because it will 
        occur for eqdsks only.
        Future dict instantiation methods will likely differ, hence the
        confusing name of this method.
```